src/util/OGCFilter.js

```diff
@@ -467,11 +524,73 @@ Ext.define('GeoExt.util.OGCFilter', {
                 return null;
             }
 
-            var childNodes = geometryNode.children || geometryNode.childNodes;
-            var serializer = new XMLSerializer();
-            var serializedValue = serializer
-                .serializeToString(childNodes[0]);
-            return serializedValue;
+            if (wfsVersion === '2.0.0') {
+                // supported geometries: Point, LineString and Polygon
+                // in case of multigeometries, the first one is used.
+                var geometryType = geometry.getType();
+                var staticMe = GeoExt.util.OGCFilter;
+                var isMulti = geometryType.indexOf('Multi') > -1;
+                switch (geometryType) {
+                case 'Polygon':
+                case 'MultiPolygon':
```

Comment:
Why is a case handling necessary for WFS 2.0.0? Couldn't it be done like for WFS v1.x (the else part)?

Reply:
I didn't find a way to add `gml` namespace prefix to GML created by openlayers `ol.format.GML3` neither using W3C DOM API nor by OpenLayers options of `GML` format . The namespaces are required in FE 2.0 AFAIK
In 2106abe017 I moved GML serialization used for WFS 1.x to `else` block to have a more clear discrimination.